### DISCUSSION TITLE
Resolve an mDNS name from a thread device

### DISCUSSION THREAD

**Discussion by fgervais** [openthread/openthread#7650] [Q&A] [answered]
Can a thread device resolve an mDNS `.local` name of a server outside the thread network i.e. on the other 
side of the border router nat64?

Something like this:

```
+-------------------+         +---------------+       +----------------+
|                   |resolve  |               |       |                |
|   Thread Device   +---------> Border Router +------->  server.local  |
|                   |         |               |       |                |
+-------------------+         +---------------+       +----------------+
```

**Comment by jwhui** [comment, marked as answer]
Yes, the OpenThread Border Router implements Discovery Proxy for Multicast DNS-Based Service Discovery [RFC 8766](https://datatracker.ietf.org/doc/html/rfc8766). This allows the Thread Border Router to discover wevices on the Wi-Fi/Ethernet side using mDNS on behalf of the Thread Device.

On the Thread Device, you can try the following command sequence:

1. Enable the DNS client to auto-configure the default server address:
    ```
    srp client autostart enable
    ```
2. Verify the DNS client's default service address:
    ```
    dns config
    ```
3. Resolve a service:
    ```
    dns resolve HOST.default.service.arpa
    ```
    Note that the Discover Proxy will map queries with `.deafult.service.arpa` to `.local` on the Wi-Fi/Ethernet side.

Hope that helps.

**Reply by fgervais** [reply]
Really nice. Thank you very much I’ll try that out.
